src/platforms/ctxlink/platform.c

```diff
@@ -330,6 +322,50 @@ bool platform_nrst_get_val(void)
 	return false;
 }
 
+void platform_read_adc(void)
+{
+	if (--read_adc_period != 0)
+		return;
+	read_adc_period = READ_ADC_PERIOD;
+	uint32_t adc_value;
+	//
+	// The ADC is read here; there are two channels read for the target
+	// and battery. A 4-phase system is used:
+	//	phase 0 -> start target voltage conversion
+	//	phase 1 -> Read and update the cached target voltage
+	//	phase 2 -> start the battery voltage conversion
+	//	phase 3 -> read and update the cached battery voltage
+	switch (adc_phase) {
+	default: {
+		//
+		// Something went wrong, reset the phase and start again
+		//
+		adc_phase = 0;
+		BMD_FALLTHROUGH;
+	}
+	case 0: {
+		adc_set_regular_sequence(ADC1, 1, &adc_channels[CTXLINK_ADC_TARGET]);
+		adc_start_conversion_regular(ADC1);
+		break;
+	}
+	case 1: {
+		adc_value = (adc_read_regular(ADC1) * 16U) / 10U;
+		__atomic_load(&adc_value, &input_voltages[CTXLINK_ADC_TARGET], __ATOMIC_RELAXED);
```

Comment:
This looks like it should be atomic_store() to write `adc_value` into `input_voltages[CTXLINK_ADC_TARGET]`? Same comment about using the `<stdatomic.h>` call. Likewise the one on line 363.

Reply:
Fixed